cmd/rpc/pool.go

```diff
@@ -0,0 +1,49 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rpc
+
+import (
+	"bytes"
+	"sync"
+)
+
+// A Pool is a type-safe wrapper around a sync.Pool.
+type Pool struct {
+	p *sync.Pool
+}
+
+// NewPool constructs a new Pool.
+func NewPool() Pool {
+	return Pool{p: &sync.Pool{
+		New: func() interface{} {
+			return &bytes.Buffer{}
+		},
+	}}
+}
+
+// Get retrieves a bytes.Buffer from the pool, creating one if necessary.
+func (p Pool) Get() *bytes.Buffer {
+	buf := p.p.Get().(*bytes.Buffer)
+	buf.Reset()
```

Comment:
`buf.Reset` has already been called in `Pool.Put`. We don't need it here too.

Reply:
Doesn't hurt to do it again.

Reply:
What is the reason behind calling `buf.Reset` in both `Get` and `Put`?

Reply:
The reason is since Get() and Put() are not synchronized, it is to avoid any unexpected situations.